### ISSUE TITLE
GET /categories/hierarchy

### ISSUE THREAD

**Issue by waynekoepcke** [ShelterTechSF/sheltertech-go#43]
Migrate GET /categories/hierarchy to GO
This endpoint queries the 'categories' table and where top_level=true, a nested query to the 'category-relationships' table finds child categories. The response body is an object with the "categories" prefix, followed by an array of empty slices for categories with no children, and sub-arrays of child objects if they exist, ordered (optionally) by rank.

**Comment by zen-gineer** [comment]
I understand this endpoint is not to be migrated. Okay to close it?

**Comment by richardxia** [comment]
Right, it looks like the frontend app doesn't hit this endpoint at all (the word "hierarchy" doesn't even appear in that repo), so I'm closing this as "not planned".
